core/xml.ts

```diff
@@ -188,8 +190,9 @@ export function blockToDom(
 
   const commentText = block.getCommentText();
   if (commentText) {
-    const size = block.commentModel.size;
-    const pinned = block.commentModel.pinned;
+    const comment = block.getIcon(COMMENT_TYPE) as CommentIcon;
```

Comment:
See here for context about icon types: https://github.com/google/blockly/pull/7115#discussion_r1211754816

Essentially, I think `CommentIcon.TYPE` is the better API, but it breaks our build.

Reply:
iiuc, this one in particular isn't likely to break the build (you also import `CommentIcon` here so I would be confused about how) but the mutator icon is probably the one that broke it and you're doing this here to be consistent. is that right? if so, that's fine, just trying to understand what's happening.

Reply:
This one also creates circular dependencies :/ But I'm not sure why.

Wrt the import of `CommentIcon`, it doesn't create circular dependencies itself b/c it's only used as a type, which means it gets removed from the tsc output. I've changed it to `import type` to make this clear.

Accessing `CommentIcon.TYPE` means that you're using it as a value, which triggers the circular dependencies.

I'm imagining this is happening now (and not before) because `Xml` is importing `CommentIcon` directly, while before `Comment` was only imported by `BlockSvg`. But beyond that I'm not sure what the exact cycle is.